tfjs-backend-webgpu/src/binary_op_util.ts

```diff
@@ -42,8 +42,9 @@ export enum BinaryOpType {
 }
 
 const CHECK_NAN_SNIPPET = `
-  if (isnan(a)) { return a; }
-  if (isnan(b)) { return b; }
+  if isNaN { return valueForNaN; }
+  if isnan(a) { return valueForNaN; }
+  if isnan(b) { return valueForNaN; }
```

Comment:
LGTM to only change above three lines to align with `CHECK_NAN_SNIPPET_VEC4`.
For other part, It seems that you are introducing more duplicated things. Like twice `checkNaNSnippet` check and some never running code. I understand you may clean the NaN ops to remove the extra `checkNaNSnippet` in following PRs. But I prefer you do it when everything is ready or do ops refactor first. I hope each PR is clean and don't bring some confusing things.

Reply:
> LGTM to only change above three lines to align with CHECK_NAN_SNIPPET_VEC4.

Will do in the final cleanup, but not possible now. Current scalar implementations are lacking the definition of `resultTemp`.